presto-testng-services/src/main/java/io/prestosql/testng/services/FlakyTestRetryAnalyzer.java

```diff
@@ -49,23 +51,31 @@
     @Override
     public boolean retry(ITestResult result)
     {
-        if (result.isSuccess()) {
-            return false;
-        }
         if (!isEnabled()) {
             log.info("not retrying; FlakyTestRetryAnalyzer disabled");
             return false;
         }
+
         Method javaMethod = result.getMethod().getConstructorOrMethod().getMethod();
         if (javaMethod == null) {
             log.info("not retrying; cannot get java method");
             return false;
         }
+
         Flaky annotation = javaMethod.getAnnotation(Flaky.class);
         if (annotation == null) {
             log.info("not retrying; @Flaky annotation not present");
             return false;
         }
+
+        if (!javaMethod.isAnnotationPresent(Test.class)) {
```

Comment:
unit test?

Reply:
with current design of the tests, idk how would this be possible

Reply:
Also, any code calling reportListenerFailure (which does hard exit), is not testable